currency-rates/internal/services/rate.go

```diff
@@ -15,27 +15,19 @@ type RateClient interface {
 
 type Rate struct {
 	logger tools.Logger
-	rateClients []RateClient
+	rateClient RateClient
 }
 
-func NewRate(logger tools.Logger, rateClients []RateClient) Rate {
-	return Rate{
+func NewRate(logger tools.Logger, rateClient RateClient) *Rate {
+	return &Rate{
 		logger: logger,
-		rateClients: rateClients,
+		rateClient: rateClient,
 	}
 }
 
-func (r Rate) GetDollarRate(ctx context.Context) (float64, error) {
-	var rate float64
-	var err error
-
-	for _, c := range r.rateClients {
+func (r *Rate) GetDollarRate(ctx context.Context) (float64, error) {
 		ctx, cancel := context.WithTimeout(ctx, apiCallTimeout)
 		defer cancel()
```

Comment:
This timeout will be in not for every call, but for all the chain. Maybe better to use timeout straight in the clients?
Here you can make global timeout of this operation, because context allows us to make inheritance

Reply:
Yes, it's a good point. I missed that, thanks for the advice